pkg/apis/eksctl.io/v1alpha5/validation_test.go

```diff
@@ -308,6 +318,23 @@ var _ = Describe("ClusterConfig validation", func() {
 			Expect(err.Error()).To(Equal("nodeGroups[1].iam.instanceRoleARN and nodeGroups[1].iam.instanceRoleName cannot be set at the same time"))
 		})
 
+		It("should not allow setting instanceRoleARN and attachPolicyARNs", func() {
+			ng1.IAM.InstanceRoleARN = "r1"
+			ng1.IAM.AttachPolicy = cft.MakePolicyDocument(
+				cft.MapOfInterfaces{
+					"Effect": "Allow",
+					"Action": []string{
+						"s3:Get*",
+					},
+					"Resource": "*",
+				},
+			)
+
+			err = api.ValidateNodeGroup(1, ng1)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("nodeGroups[1].iam.instanceRoleARN and nodeGroups[1].iam.attachPolicy cannot be set at the same time"))
```

Comment:
@adammw For the record and the future. We are moving away from this check err, then compare the `.Error()` to MatchError with ContainsSubstring since `MatchError` does a nil check. So if a file contains those two, we replace them with one. MatchError was added later.